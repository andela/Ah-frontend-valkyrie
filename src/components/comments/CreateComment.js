import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import 'react-toastify/dist/ReactToastify.css';
import { addComment } from './actions/actions';

export class CreateComment extends Component {
  constructor( props ) {
    super( props );
    this.state = {
      newCommentBody: '',
    };
  }

  handleOnChange = ( event ) => {
    this.setState( {
      newCommentBody: event.target.value,
    } );
  }

  handleOnSubmit = ( event ) => {
    event.preventDefault();
    const { articleSlug } = this.props;
    const { newCommentBody } = this.state;
    if ( event.target.value !== '' ) {
      const comment = {
        comment: {
          body: newCommentBody,
        },
      };
      this.props.addComment( articleSlug, comment );
    }
    this.resetTextArea();
  }

  resetTextArea = () => {
    this.setState( { newCommentBody: ''} )
  }

  render() {
    const { newCommentBody } = this.state;
    return (
      <div>
        <form onSubmit={ this.handleOnSubmit }>
          <div className="form-group">
            <textarea 
              name="commentBox" 
              onChange={ this.handleOnChange } 
              rows="3" 
              className="form-control" 
              value={ newCommentBody }
              placeholder="Add comment....."
            />
          </div>
          <input type="submit" value="Add comment" className="btn btn-sm btn-primary"/>
        </form>
        {
        }
      </div>
    );
  }
}

CreateComment.propTypes = {
  articleSlug: PropTypes.string.isRequired,
  addComment: PropTypes.func.isRequired,
};

const mapStateToProps = state => ( {
  comments: state.comments.comments,
} );

export default connect( mapStateToProps, { addComment } )( CreateComment );