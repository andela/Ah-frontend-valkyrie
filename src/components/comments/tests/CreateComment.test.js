import React from "react";
import { shallow } from "enzyme";
import { CreateComment } from "../CreateComment";

describe("<Create comment component />", () => {
  let mockComponent;
  const props = {
    articleSlug: "new-article-slug",
    addComment: jest.fn(),
  };
  const event = {
    target: {
      value: "what a value",
    },
  };
  beforeEach(() => {
    mockComponent = shallow(<CreateComment {...props} />);
  });
  it("renders as expected", () => {
    expect(mockComponent).toMatchSnapshot();
  });
  it("handle onChange", () => {
    mockComponent.instance().setState = jest.fn();

    mockComponent.instance().handleOnChange(event);

    expect(mockComponent.instance().setState).toBeCalled();
  });

  it("handle submit", () => {
    const event = {
      target: {
        value: "another value",
      },
      preventDefault: jest.fn(),
    };
    mockComponent.setProps({
      event: {
        target: {
          value: "radnom commen",
        },
      },
    });

    mockComponent.instance().handleOnSubmit(event);
    expect(props.addComment).toBeCalled();
  });
});
