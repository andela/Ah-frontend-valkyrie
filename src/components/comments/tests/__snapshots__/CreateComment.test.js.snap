// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Create comment component /> renders as expected 1`] = `ShallowWrapper {}`;
